### ISSUE TITLE
Library doesn't work outside of Unity

### ISSUE THREAD

**Issue by TheHans255** [camsjams/i18n-unity-csharp#4]
While the README claims that this internationalization library can be used as a standalone library for other applications, the library itself uses some Unity types (`UnityEngine`, `Debug`, `Text`, a few others). Would it be possible to use some preprocessor directives (like `#ifdef UNITY`) in order to use different types?

**Comment by GabLeRoux** [comment]
Yup, you're right, I haven't tried outside of Unity, but from the current code, it can't work right now since it uses some Unity3D classes such as `Debug` and `TextAsset`.

You can give a try to my [`feature/remove-unity-dependency branch`](https://github.com/GabLeRoux/i18n-unity-csharp/tree/feature/remove-unity-dependency)

I tried what you suggested here, but haven't tried the project somewhere else than inside Unity. If you'd like to give this a try, just let us know. Feel free to send a Pull-Request and instructions on how to use it outside of Unity and I'll be happy to give it a try.

**Comment by TheHans255** [comment]
Thanks for doing this. However, your changes in that feature branch don't work for two reasons:
1) `UNITY_EDITOR` is only defined while running inside the Editor. This means that the Unity library sections are taken out when the project is actually built. You would need a different `#define` that would catch all cases when Unity is used.
2) You don't provide any `#else` blocks with alternate implementations - the idea is that when you're not in Unity, you would use System types to do the same thing. Right now, when not in Unity, the program actually does absolutely nothing.

**Comment by GabLeRoux** [comment]
Sure, that was only to give you a starting point in case you would like to make it happen. I did not try to build it, but sure thing, `UNITY_EDITOR` is not what we're looking for here. I did not try `UNITY` either, maybe it works. [I couldn't find it in documentation](https://docs.unity3d.com/Manual/PlatformDependentCompilation.html)

If I had to use i18n in a non-unity .net project, I would probably go with something more common / tested such as one of these:

* https://github.com/turquoiseowl/i18n
* https://github.com/jeffijoe/MessageFormat.NET
* https://github.com/tom-englert/ResXResourceManager
* https://github.com/neris/NGettext

**Comment by camsjams** [comment]
For history, it _did_ but we had a project that was giving us trouble so I had to add it in the file as a dependency:
https://github.com/MoonGateLabs/i18n-unity-csharp/commit/0be41e208efca92c31c128fba8c76f7a520715e3
